lib/streamlit/error_util.py

```diff
@@ -50,7 +51,21 @@ def handle_uncaught_app_exception(e: BaseException) -> None:
         # Use LOGGER.error, rather than LOGGER.debug, since we don't
         # show debug logs by default.
         LOGGER.error("Uncaught app exception", exc_info=e)
-        st.error(_GENERIC_UNCAUGHT_EXCEPTION_TEXT)
+        stacktrace_list = traceback.format_exc().splitlines()
+        # remove the exception message so we can keep information private
+        filtered_stacktrace_list = []
+        while stacktrace_list:
+            entry = stacktrace_list.pop()
+            if "Traceback (most recent call last):" in entry:
+                break
+            filtered_stacktrace_list.insert(0, entry)
+        # filtered_stacktrace_list = [entry for entry in stacktrace_list if not _is_in_streamlit_package(entry)]
+        filtered_stacktrace_list.pop()
```

Comment:
I think you should be able to simplify this greatly by writing
```python
filtered_stacktrace_list = stacktrace_list[start:end]
```

to drop the lines that may potentially contain sensitive information (I think they should always be in the same place?)